src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,63 @@
+import classNames from 'classnames';
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAppDispatch } from '../../app/hooks';
+import { setAppliedQuery } from '../../features/querySlice';
+import { SearchIconColor } from '../../types/SearchIconColor';
+import Icon from '../Icon/Icon';
+import styles from './SearchBar.module.scss';
+
+type Props = {
+  query: string;
+  setQuery: (value: string) => void;
+  applyQuery?: (value: string) => void;
+  searchIconColor: SearchIconColor;
+};
+
+export const SearchBar: React.FC<Props> = ({
+  query,
+  setQuery,
+  applyQuery,
+  searchIconColor,
+}) => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const onHandleQueryInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+
+    if (applyQuery) {
+      applyQuery(event.target.value);
+    }
+  };
+
+  const onHandleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!applyQuery) {
+      dispatch(setAppliedQuery(query));
+      navigate('/products');
+    }
+  };
```

Comment:
Redirect is still working on product page. I think it doesn't affect anything but y can validate it to prevent unpredictable behavior in future.

Reply:
in my case, it doesn't work. i've tried